inspirehep/modules/records/api.py

```diff
@@ -91,13 +101,23 @@ def update(self, *args, **kwargs):
                 files for the documents and figures from this record's files
                 iterator before downloading them, for example to merge existing
                 records.
+            skip_files(bool): if ``True`` it will skip the files retrieval
+                described above. Note also that, if not passed, it will fall
+                back to the value of the ``RECORDS_SKIP_FILES`` configuration
+                variable.
+
         """
+        config_skip_files = current_app.config.get('RECORDS_SKIP_FILES')
+
         files_src_records = kwargs.pop('files_src_records', ())
-        super(InspireRecord, self).update(*args, **kwargs)
-        self.download_documents_and_figures(
-            only_new=True,
-            src_records=files_src_records,
-        )
+        skip_files = kwargs.pop('skip_files', config_skip_files)
+
+        record = super(InspireRecord, self).update(*args, **kwargs)
+
+        if not skip_files:
+            record.download_documents_and_figures(src_records=files_src_records, only_new=True)
+
+        return record
```

Comment:
Well, here I made a silly mistake. I'm going to fix it tomorrow, as I only needed the `create` method to work for the HAL push.